native/native.tests/tests/org/jetbrains/kotlin/konan/test/blackbox/CompilerOutputTest.kt

```diff
@@ -109,6 +110,32 @@ abstract class CompilerOutputTestBase : AbstractNativeSimpleTest() {
         KotlinTestUtils.assertEqualsToFile(goldenData, compilationResult.toOutput())
     }
 
+    @Test
+    fun testObjCExportDiagnosticsErrors() {
+        Assumptions.assumeTrue(targets.hostTarget.family.isAppleFamily)
+
+        val rootDir = File("native/native.tests/testData/compilerOutput/ObjCExportDiagnostics")
+        val settings = testRunSettings
+        val lib1 = compileLibrary(settings, rootDir.resolve("lib1.kt")).assertSuccess().resultingArtifact
+        val lib2 = compileLibrary(settings, rootDir.resolve("lib2.kt")).assertSuccess().resultingArtifact
+
+        val freeCompilerArgs = TestCompilerArgs(
+            "-Xinclude=${lib1.path}",
+            "-Xinclude=${lib2.path}",
+            "-Xbinary=objcExportErrorOnNameCollisions=true"
+        )
+        val expectedArtifact = TestCompilationArtifact.ObjCFramework(buildDir, "testObjCExportDiagnosticsErrors")
+
+        val compilationResult = ObjCFrameworkCompilation(
+            settings,
+            freeCompilerArgs,
+            sourceModules = emptyList(),
+            dependencies = emptyList(),
+            expectedArtifact
+        ).result
+        assertIs<TestCompilationResult.Failure>(compilationResult)
```

Comment:
It would be nice to also check the actual output. E.g. to make sure that all errors are present as expected.